### ISSUE TITLE
DASHBOARD: Card title turns into question title if there is no results for the chart

### ISSUE THREAD

**Issue by alantygel** [metabase/metabase#7948]
Hi,

I noticed a strange behavior which I think is a bug. 

I have a dashboard with several questions, all of them dependent on an argument (same for all questions). In all questions, I had to edit the card title in order to describe the source of data (this is related to this feature request: #6434).

This workaround solved my issue, however I've noticed that, if the question presents no data, the card title appears as question title. In my point of view, this is a bug.

- Your browser and the version: Firefox 60.0.2 (64-bit)
- Your operating system: Xubuntu 16.04
- Your databases: MySQL
- Metabase version: 0.29.3
- Metabase hosting environment: Linux/Ubuntu 16.04
- Metabase internal database: H2 (default)

Repeatable steps to reproduce the issue
* Create a question with a variable. Depending on the value of this variable, the question may present a result or not
* Add this question to a dashboard and configure the variable to be a dashboard filter
* Change the card title to something different from the question title
* Then, if for the given filter there is a result, the card title show as expected; otherwise, the question title shows.

Screenshots bellow presents both situation for the same card; first, with results, and the correct card title; then without results, with the wrong card title.

![dashboard-result1](https://user-images.githubusercontent.com/1626935/41974557-ad763540-79ee-11e8-8f5a-d739e14256a2.png)
![dashboard-noresult](https://user-images.githubusercontent.com/1626935/41974558-ad988744-79ee-11e8-95f1-4769f91e0d97.png)

**Comment by riccardocagnasso** [comment]
I can confirm this

**Comment by notrom** [comment]
Metabase v0.32.9 and am seeing a similar thing too.

I've found that if the question is a Table then the title is displayed correctly, either the title from the base question or if provided, the title from the dashboard card.

But if I change the question to be a bar chart it reverts the measure name i.e. "Count", can't change this via the card settings.

Using the Sample database:
Here's the Question as a table (there is no data in that date range):
![image](https://user-images.githubusercontent.com/4504437/59890160-bef4ef00-9423-11e9-8989-865a9b4bf230.png)

Dashboard using that question - title is updated in the dashboard and showing correctly:
![image](https://user-images.githubusercontent.com/4504437/59890184-e0ee7180-9423-11e9-8215-b55d9384587b.png)

Changed Question to be a bar chart, no other changes, title on card has been changed to "Count" which is neither the card title in the dashboard or the question title:
![image](https://user-images.githubusercontent.com/4504437/59890217-05e2e480-9424-11e9-95ec-4436cbf8e350.png)

Changed the measure to a Sum in the question, and the card title now shows as "Sum".

Setting the question to be a scatter plot, the dashboard title again displays as expected.
